### PR TITLE
Features/user local identity

### PR DESCRIPTION
**Pull request by gdchamal**
Important change to come

An user can have many local identity (address where he can receive messages)

An user name is no more an email by default, but have a local identity using a default domain as mail on a Caliopen platform acting for receiving email.

We must discuss about user name syntax (again but this time a regex should be defined) and others rules (unicity on lowercase ?)

ping @iamdey @sapiens-sapide

### CODE REVIEW

**Comment by sapiens-sapide** [comment]
it seems this PR breaks the application. When trying to auth on /autentications I got the follow error in the application console (stripped to last lines) : 
```
  File "cassandra/cluster.py", line 1710, in cassandra.cluster.Session.execute (cassandra/cluster.c:30976)
  File "cassandra/cluster.py", line 3343, in cassandra.cluster.ResponseFuture.result (cassandra/cluster.c:68510)
InvalidRequest: Error from server: code=2200 [Invalid query] message="Undefined column name local_identities"
2016-11-22 17:31:20,645 INFO  [caliopen_api.user.user][waitress] Authentication error for dev@caliopen.local : Error from server: code=2200 [Invalid query] message="Undefined column name local_identities"
```

**Comment by gdchamal** [comment]
Yes cassandra schema change, need to run again the caliopen setup command, but I'm not sure all upgrade cases will be covered with this command.

**Comment by iamdey** [comment]
started a page for the username format: https://pad.caliopen.org/p/username

**Comment by gdchamal** [comment]
Hence the username syntax and validation is defined I wll finish this feature

**Comment by gdchamal** [comment]
Ready for merge with username syntax

File: src/backend/main/py.main/caliopen_main/user/core/user.py [hunk 1]
```diff
@@ -197,10 +197,11 @@ def rollback_username_storage(username):
             raise ValueError("new user malformed")
 
         try:
+            recovery = new_user.recovery_email
```
Comment (gdchamal) [inline src/backend/main/py.main/caliopen_main/user/core/user.py]: pep8 rule, the line was too long ..

File: src/backend/protocols/py.smtp/caliopen_smtp/agent.py [hunk 2]
```diff
@@ -29,7 +29,7 @@ def process_user_mail(self, user, raw_msg_id):
     def resolve_users(self, rpcts):
         users = []
         for rcpt in rpcts:
-            user = User.by_name(rcpt)
+            user = User.by_local_identity(rcpt)
             users.append(user)
         return users
```
Comment (gdchamal) [inline src/backend/protocols/py.smtp/caliopen_smtp/agent.py]: Should fit better in your current work when this branch will be merged

**Pull request merged**